### PR TITLE
Feat/add email verification

### PR DESCRIPTION
**Pull request by cmazzullo**
- Added an endpoint at /api/v1/verify/ that takes a POST of ('user_id', 'verification_key') which verifies + activates a user if the key is correct.
- Now we send an email upon user registration containing their verification link
- Assumed that this URL: https://covexit.de/verify/MYKEYHERE will be set up on the frontend for users to see + to send the POST request to the backend
- Wrote a couple tests, added model_bakery as a dep to make testing more convenient

### CODE REVIEW

File: backend/covexit/account/models.py [hunk 1]
```diff
@@ -1,7 +1,43 @@
...
+
+def send_verification_email(user):
+    link = create_verification_link(user)
+    send_mail(
+        'Covexit Email Verification',
+        VERIFICATION_MESSAGE.format(link),
+        settings.DEFAULT_FROM_EMAIL,
+        [user.email],
+        fail_silently=False,
+    )
```
Comment (vellip) [inline backend/covexit/account/models.py:40]: Should we do a proxy model to User so we have this all directly on the User instance? Or does it not work that way?

File: backend/covexit/account/models.py [hunk 2]
```diff
@@ -1,7 +1,43 @@
...
 from django.utils.translation import ugettext_lazy as _
 from django.core.exceptions import ValidationError
+from django.core.mail import send_mail
+from django.conf import settings
+
+VERIFICATION_MESSAGE = """Thank you for signing up for Covexit!
+
+To activate your account, please visit this link:
+{}
+"""
```
Comment (vellip) [inline backend/covexit/account/models.py:15]: Needs to be translateable

File: backend/covexit/account/models.py [hunk 3]
```diff
@@ -1,7 +1,43 @@
...
+
+VERIFICATION_KEY_LENGTH = 30
+# Note: this will have to be set up on the frontend:
+VERIFICATION_URL = 'https://covexit.de/verify/'
```
Comment (vellip) [inline backend/covexit/account/models.py:19]: Can we get the host name from somewhere so it also works locally or is that not a good a idea?

**Comment by cmazzullo** [comment]
Addressed those points, you may need to remake your DB after the most recent commit because switching to a custom User model messed with some existing migrations

**Comment by vellip** [comment]
@cmazzullo Ahh, I didn't mean a custom User model, I meant a proxy model like so https://simpleisbetterthancomplex.com/tutorial/2016/07/22/how-to-extend-django-user-model.html#proxy 

Or do you think we are better off with a custom one at this point?

**Pull request merged**